### DISCUSSION TITLE
RPi Pico mip not installed

### DISCUSSION THREAD

**Discussion by mrks89** [micropython/micropython#11126] [RP2040 / Pico] [answered]
There is no mip on the rpi pico (without Wifi). So how would I install packages or use mip?

**Comment by bidrohini** [comment]
Here is a similar issue that you can look at: https://github.com/coin-or/python-mip/issues/276

**Comment by Josverl** [comment, marked as answer]
Install mpremote on your pc/host
Then use that to install packages to the MCU 
``` bash
pip install mpremote
mpremote mip install my_package
```

**Reply by mrks89** [reply]
Sounds great, but did not work:

```
C:\Users\xxx> mpremote connect id:E6616408433B1E34 mip install network
Install network
Installing networks (latest) from https://micropython.org/pi/v2 to /lib
Package may be partially installed
mpremote: Package not found: https://micropython.org/pi/v2/package/6/network/latest.json
```
I have to work on windows. I installed mpremote with pip, identified the rpi pico with `mpremote connect list` and connected to the pico via its id.

**Reply by robert-hh** [reply]
So a package `network` does not seem to exist. Besides that: why do you want to install it on a rpi pico without network interfaces?

**Reply by mrks89** [reply]
should exist though. At least it is used here e.g.: [git example](https://github.com/bleakview/rp_pico_w_aws_iot/blob/master/awsiot.py)
legit comment, but I have attached a sim7080 pico hat to connect via cat-m and was hoping to use the network package same as in the tutorial mentioned.

**Reply by robert-hh** [reply]
The network module exists only on boards with networking support in the hardware and cannot be loaded to other boards. The sample you linked uses obviously such a board. You should be able to use a sim7080 e.g. by UART by sending the proper AT commands and exchanging data, but that would be completely different to the example. There was some discussion for such a method in the forum about setting up a PPP connection. But I did not follow it.

**Reply by mrks89** [reply]
concerning the network package you say that won't work even with the sim module, okay. But that is a special package, what about e.g. numpy, or any other package. And another special package umqtt, won't that also work with my setup?

**Reply by robert-hh** [reply]
Numpy is a huge package, which requires way more RAM than the ~200k a RPi Pico has. It runs on a RPi3 or 4 with 10000 to 20000 times the RAM size a  RPI Pico has. There is a small similar software call ulab, but that has to be built into the firmware.  You can install umqtt.robust or umqtt.simple with mip. But umqtt uses sockets, and sockets are available together with the network module. You can tell which modules arte available in the firmware by calling:
`help("modules")`
at the REPL prompt. if you want to do networking, get e.g. a Pico W module.

**Reply by mrks89** [reply]
okay. I did not understand all this. I have a clearer picture now. Thanks a lot.
